dataclasses_json/mm.py

```diff
@@ -318,13 +320,26 @@ def schema(cls, mixin, infer_missing):
                 options['data_key'] = metadata.letter_case(field.name)
 
             t = build_type(type_, options, mixin, field, cls)
+            _account_for_decoder_if_necessary(field, t)
```

Comment:
```suggestion
            if field.metadata.get('dataclasses_json', {}).get('decoder'):
                t._deserialize = lambda v, *_a, **_kw: v
```

Also maybe add comment why a private field is being mutated here?